_build/rewriteUrl.js

```diff
@@ -0,0 +1,22 @@
+// Simple middleware that will rewrite grist-plugin-api URL.
+// We assume that Custom Widget is importing grist-plugin-api.js by including script tag:
+// <script src="https://docs.getgrist.com/grist-plugin-api.js"></script>
+module.exports = function (req, res, next) {
+  // Rewrite only for GET method and an URL containing .htm fragment.
+  if (req.url && req.url.includes(".htm") && req.method === 'GET') {
+    // Overwrite Response internal _writeRaw method.
+    const _writeRaw = res._writeRaw.bind(res);
+    res._writeRaw = function (data, encoding, callback) {
+      // Make sure that the Response data is a string.
+      if (typeof data === 'string' && data) {
+        // Replace first occurrence of grist-plugin-api.js URL.
+        data = data.replace(
+          'https://docs.getgrist.com/grist-plugin-api.js',
+          'http://localhost:8080/grist-plugin-api.js    ' // extra space for length match
```

Comment:
hmm, having used this, it is very handy - but grist-core is 8484, and I'd like this to work for everyone, and be easiest to use by grist-core users.

Reply:
I added this extra space because I'm not modifying Content-Length header (will update the comment), and this is rewriting the response just before it is sent to the client.

Reply:
Ah, makes sense :+1: